### PR TITLE
Add_Remote_Rbac_Verification

### LINKED ISSUE #3: Add Rbac Remote Verification

### PR DESCRIPTION
**Pull request by quophyie**
Fixes #3

### CODE REVIEW

File: examples/server.js [hunk 1]
```diff
@@ -65,10 +69,32 @@ server.get('/some-rbac', auth.express.authenticate(), rbacExpress.allow(['update
...
 })
-server.use(rbacExpress.initialize(opts))
+// server.use(rbacExpress.initialize(opts))
+rbacExpress.initialize(opts)
 server.get('/some-unreachable', auth.express.authenticate(), (req, res) => {
   res.send({ response: 'this url should be unreachable as there are no permissions set on the route' })
 })
-server.listen(9000, () => {
+
+server.post('/some-remote-authenticated-route', auth.express.authenticate(), rbacExpress.allow(['update'], { useRemoteAuthorization: true, remoteAuthorizationEndPoint: `localhost:${PORT}/authorize` }), function (req, res, next) {
```
Comment (andrezzoid) [inline examples/server.js]: Do we need to set `{ useRemoteAuthorization: true, remoteAuthorizationEndPoint: `localhost:${PORT}/authorize` }` on every endpoint? Isn't it possible to define it globally on the RBAC configuration? I think that'd be pretty cool.

File: lib/express/index.js [hunk 2]
```diff
@@ -3,18 +3,125 @@
...
 const check = require('check-types')
+const Enums = require('./../enum/index')
 let rbacBase = null
-// var onHeaders = require('on-headers')
-var interceptor = require('express-interceptor')
-// var tamper = require('tamper')
-// var bafMiddleware = require('before-and-after')
+const interceptor = require('express-interceptor')
+const _ = require('lodash')
+const request = require('request')
```
Comment (andrezzoid) [inline lib/express/index.js]: There's a `request-promise` package so you can write promise-based requests, if you'd like.

File: examples/server.js (hunk 1 above)
Comment (quophyie) [inline examples/server.js]: @andrezzoid No  we dont need to set it on every route. You can set it globally by passing the same configuration to the `opts` object that is passed in `rbacExpress.initialize(opts)`. However, you can also set it at route level as is currently demonstrated

File: lib/express/index.js (hunk 2 above)
Comment (quophyie) [inline lib/express/index.js]: Actually, thats a good idea. I updated the code to use request-promise now

**Pull request merged**